bpo-40280: WASM docs and smaller browser builds (GH-32412) (#32412)
python/cpython: Tools/wasm/README.md

## wasm32-emscripten build

> Cross compiling to wasm32-emscripten platform needs the

Reviewer: ```suggestion
Cross compiling to the wasm32-emscripten platform needs the
```

- pthread support requires WASM threads and SharedArrayBuffer (bulk memory).
  The runtime keeps a pool of web workers around. Each web worker uses
  several file descriptors (eventfd, epoll, pipe).

# Hosting Python WASM builds

The simple REPL terminal uses SharedArrayBuffer. For security reasons
browsers only provide the feature in secure environents with cross-origin

> isolation. The webserver must sent cross-origin headers and correct mimetypes

Reviewer: ```suggestion
isolation. The webserver must send cross-origin headers and correct MIME types
```

for the JavaScript and WASM files. Otherwise the terminal will fail to load

> an error message like ``Browsers disable shared array buffer``.

Reviewer: ```suggestion
with an error message like ``Browsers disable shared array buffer``.
```
